greedy-timetabling/src/main/java/org/acme/schooltimetabling/rest/TimeTableResource.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.acme.schooltimetabling.rest;
+
+import io.quarkus.panache.common.Sort;
+import org.acme.schooltimetabling.domain.Lesson;
+import org.acme.schooltimetabling.domain.Room;
+import org.acme.schooltimetabling.domain.TimeTable;
+import org.acme.schooltimetabling.domain.Timeslot;
+import org.acme.schooltimetabling.persistence.LessonRepository;
+import org.acme.schooltimetabling.persistence.RoomRepository;
+import org.acme.schooltimetabling.persistence.TimeslotRepository;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import java.util.Objects;
+
+@Path("timeTable")
+public class TimeTableResource {
+    public static final Long SINGLETON_TIME_TABLE_ID = 1L;
+    @Inject
+    TimeslotRepository timeslotRepository;
+    @Inject
+    RoomRepository roomRepository;
+    @Inject
+    LessonRepository lessonRepository;
+
+    // To try, open http://localhost:8080/timeTable
+    @GET
+    public TimeTable getTimeTable() {
+        return findById(SINGLETON_TIME_TABLE_ID);
+    }
+
+    @GET
+    @Path("solved")
+    public TimeTable getTimeSolvedTable() {
+
+        TimeTable timeTable = findById(SINGLETON_TIME_TABLE_ID);
+
+        for (Lesson lesson : timeTable.getLessonList()) {
+            int bestScore = Integer.MIN_VALUE;
```

Comment:
looks good. I'd rename bestScore to bestHardScore.

I wouldn't deal with soft constraints in the greedy or brute force code, we can talk about soft constraints at the end and maybe show them in the live coding